drivers/timer/silabs_sleeptimer_timer.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <zephyr/init.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/logging/log.h>
+
+#include <sl_atomic.h>
+#include <sl_sleeptimer.h>
+
+LOG_MODULE_REGISTER(silabs_sleeptimer_timer);
+
+/* Maximum time interval between timer interrupts (in hw_cycles) */
+#define MAX_TIMEOUT_CYC (UINT32_MAX >> 1)
+#define MIN_DELAY_CYC   (4U)
+
+#define DT_RTC DT_COMPAT_GET_ANY_STATUS_OKAY(silabs_gecko_stimer)
+
+/* Ensure interrupt names don't expand to register interface struct pointers */
+#undef RTCC
```

Comment:
I don't see any reference to RTCC in this file.
(That's said this symbol is super intrusive and it should be fixed in the upstream)

Reply:
Yeah, it's annoying. The sleeptimer HAL requires us to connect the <peripheral>_IRQHandler function for the peripheral backing the implementation to the interrupt vector table, since Zephyr doesn't use the standard CMSIS vector table. Unfortunately, this function has a peripheral-specific name, rather than a generic alias like `sl_sleeptimer_irq_handler`.

The "least bad" way of doing this I came up with (while avoiding HAL-isms in DT) was to encode the interrupt name in the `interrupt-names` DT property, and use `CONCAT(DT_IRQ(...), _IRQHandler)` to construct the correct function name. However, this causes unwanted macro expansion for `RTCC` to the CMSIS struct pointer for the peripheral block. SYSRTC is unaffected, since its peripheral macro is `SYSRTC`, but its interrupt name is `SYSRTC_APP`. Hence the single `#undef`, since as you note, we don't actually want or need the RTCC struct pointer in this driver.

We should work with upstream to get the interrupt handler aliased to `sl_sleeptimer_irq_handler`, similar to how the HFXO manager driver already has a generic `sl_hfxo_manager_irq_handler` symbol for its interrupt handler.